### PR TITLE
Allow hiding "Keep formatting" in "dont_override" config

### LINKED ISSUE #9703: Allow hiding "Keep formatting" in "dont_override" config
### Prerequisites

- [X] I have [searched](https://github.com/roundcube/roundcubemail/issues?q=is%3Aissue) for duplicate or closed feature requests
- [X] I have [searched](https://plugins.roundcube.net/) for plugins that provide already provide the feature

### Proposal

I'd love to hide the checkbox "Keep formatting" on the email compose page.

### Motivation and context

According to [this blog entry](https://kolabian.wordpress.com/2021/11/07/a-few-small-improvements/), it is just for power users and therefore it would be nice to be removable.

### PR DESCRIPTION
**Pull request by JE4GLE**
Fixes #9703 
According to [this blog entry](https://kolabian.wordpress.com/2021/11/07/a-few-small-improvements/), "Keep formatting" is for power users and therefore it would be nice to be removable.

### CODE REVIEW

**Comment by pabzm** [comment]
Thank you for the PR! Please rebase onto the latest state of the "master" branch and remove the translation-related commits.

**Comment by alecpl** [comment]
This is incomplete. Well, maybe a rare case, but what if you set `keep_formatting=true`, and then add it to `dont_override`? A small change in `rcmail_action_mail_send::run()` is needed to support this case.

File: program/actions/mail/send.php [hunk 1]
```diff
@@ -47,6 +47,10 @@ public function run($args = [])
 
         $saveonly = !empty($_GET['_saveonly']);
         $savedraft = !empty($_POST['_draft']) && !$saveonly;
+        $keepformatting = !empty($_POST['_keepformatting']);
+        if (in_array('keep_formatting', $rcmail->config->get('dont_override', []))) {
+            $keepformatting = false;
```
Comment (alecpl) [inline program/actions/mail/send.php:52]: This is wrong. It should be 
```php
if (in_array('keep_formatting', $rcmail->config->get('dont_override', []))) {
    $keepformatting = !empty($rcmail->config->get('keep_formatting'));
} else {
    $keepformatting = !empty($_POST['_keepformatting']);
}
```

**Review by alecpl — CHANGES_REQUESTED** [review]

File: program/actions/mail/send.php (hunk 1 above)
Comment (pabzm) [reply program/actions/mail/send.php:52]: Exactly my thought. The configured value must be honoured. 👍

Comment (JE4GLE) [reply program/actions/mail/send.php]: Thank you, I didn't know, there was a configurable value.

Comment (alecpl) [reply program/actions/mail/send.php]: Oops. I thought there was, but you're right. However, I'd leave it as-is. We may introduce the option in future.

**Review by pabzm — APPROVED** [review]
Looks good to me!

@JE4GLE  We need a line in the Changelog, would you maybe add one?

**Comment by pabzm** [comment]
/remind me to merge on Dec 4th

File: program/actions/mail/send.php [hunk 2]
```diff
@@ -47,6 +47,11 @@ public function run($args = [])
 
         $saveonly = !empty($_GET['_saveonly']);
         $savedraft = !empty($_POST['_draft']) && !$saveonly;
+        if (in_array('keep_formatting', $rcmail->config->get('dont_override', []))) {
```
Comment (cebe) [inline program/actions/mail/send.php:50]:
```suggestion
        if (in_array('keep_formatting', $rcmail->config->get('dont_override', []), true)) {
```

**Review by cebe — APPROVED** [review]

File: program/actions/mail/send.php (hunk 2 above)
Comment (pabzm) [reply program/actions/mail/send.php]: To clarify: I know what the change technically does. But I don't appreciate someone jumping in from the sidelines without communicating. Please respect the development work and talk to us.

**Review by pabzm — CHANGES_REQUESTED** [review]
I'd prefer to mention the actual config mechanism in the Changelog

**Review by pabzm — APPROVED** [review]
Thank you very much! 👍

**Pull request closed without merging**